pyfixest/estimation/literals.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Literal, get_args
+
+prediction_type = Literal["response", "link"]
+vcov_type_options = Literal["iid", "hetero", "HC1", "HC2", "HC3"]
+weights_type_options = Literal["aweights", "fweights"]
+fixef_rm_options = Literal["singleton", "none"]
+solver_options = Literal["np.linalg.solve", "np.linalg.lstsq"]
+
+
+def validate_literal_argument(arg: Any, literal: Any) -> None:
+    """
+    Validate if the given argument matches one of the allowed literal types.
+
+    This function checks whether the provided `arg` is among the valid types
+    returned by `get_args(literal)`. If not, it raises a ValueError with an
+    appropriate error message.
+
+    Parameters
+    ----------
+    arg : Any
+        The argument to validate.
+    literal : Any
+        A Literal type that defines the allowed values for `arg`.
+
+    Raises
+    ------
+    ValueError
+        If `arg` is not one of the valid types defined by `literal`.
+    """
+    valid_types = get_args(literal)
```

Comment:
I'm wondering if we have to catch the case where `valid_types` is empty? This is because the function argument `literal` can be `Any`. If `literal` is a type without type arguments, e.g., `get_args(int)` returns an empty tuple.

Reply:
My preference would be that the function raises an error if `valid_types` is empty so that one should only call the validation function on a non-empty literal type.

This would also catch the case where `literal` is not a `typing.Literal` object (previous comment) so that we don't have to worry about more specific type hints?

Reply:
I now raise an error if the literal argument is not _LiteralGenericAlias. This seems to match the use case e.g. when I define EmptyType: Literal and feed it as the literal argument to _validate_literal_type